src/software/thunderscope/robot_diagnostics/robot_view.py

```diff
@@ -17,168 +15,44 @@ class RobotView(QWidget):
-            4: [self.pink, self.pink, self.pink, self.green],
-            5: [self.pink, self.green, self.pink, self.green],
-            6: [self.green, self.green, self.pink, self.green],
-            7: [self.green, self.pink, self.pink, self.green],
-            8: [self.green, self.green, self.green, self.green],
-            9: [self.pink, self.pink, self.pink, self.pink],
-            10: [self.pink, self.pink, self.green, self.green],
-            11: [self.green, self.green, self.pink, self.pink],
-            12: [self.pink, self.green, self.green, self.green],
-            13: [self.pink, self.green, self.pink, self.pink],
-            14: [self.green, self.pink, self.green, self.green],
-            15: [self.green, self.pink, self.pink, self.pink],
-        }
-
         self.layout = QVBoxLayout()
 
-        self.robot_layouts = [QHBoxLayout() for x in range(MAX_ROBOT_IDS_PER_SIDE)]
-        self.robot_status_layouts = [
-            QVBoxLayout() for x in range(MAX_ROBOT_IDS_PER_SIDE)
+        self.robot_info_widgets = [
+            RobotInfo(x, load_fullsystem) for x in range(MAX_ROBOT_IDS_PER_SIDE)
         ]
-        self.robot_battery_progress_bars = [
-            QProgressBar() for x in range(MAX_ROBOT_IDS_PER_SIDE)
-        ]
-        self.breakbeam_labels = [QLabel() for x in range(MAX_ROBOT_IDS_PER_SIDE)]
-
-        self.other_label = [QLabel() for x in range(MAX_ROBOT_IDS_PER_SIDE)]
-
-        if load_diagnostics:
-            self.robot_checkboxes = [QCheckBox() for x in range(MAX_ROBOT_IDS_PER_SIDE)]
 
         for id in range(MAX_ROBOT_IDS_PER_SIDE):
-            QVBoxLayout()
-            self.robot_battery_progress_bars[id].setMaximum(100)
-            self.robot_battery_progress_bars[id].setMinimum(0)
-            self.robot_battery_progress_bars[id].setValue(10)
-
-            self.breakbeam_labels[id].setText("BREAKBEAM")
-            self.breakbeam_labels[id].setStyleSheet("background-color: blue")
-
-            self.robot_status_layouts[id].addWidget(
-                self.robot_battery_progress_bars[id]
-            )
-            self.robot_status_layouts[id].addWidget(self.breakbeam_labels[id])
-
-            self.robot_layouts[id].addWidget(
-                self.create_vision_pattern_label(id, "b", 25)
-            )
-            self.robot_layouts[id].addLayout(self.robot_status_layouts[id])
-
-            if load_diagnostics:
-                self.robot_layouts[id].addWidget(self.robot_checkboxes[id])
-
-                # connect checkboxes to emit a toggle connection signal with the corresponding robot ID
-                self.robot_checkboxes[id].stateChanged.connect(
-                    lambda state, robot_id=id: self.toggle_robot_connection_signal.emit(
-                        robot_id
-                    )
+            self.robot_info_widgets[id].toggle_one_connection_signal.connect(
+                lambda mode, robot_id: self.toggle_all_connection_signal.emit(
```

Comment:
Curious if this can be simplified. One signal triggering another signal

Reply:
So the reason its like this is cuz thunderscope_main can only access the thunderscope object. I think one better way would be to pass in the robot_view's signal to each robot_info widget, and they can just emit this signal